src/server/pachyderm_test.go

```diff
@@ -10057,6 +10056,89 @@ func TestListTag(t *testing.T) {
 	require.Equal(t, "Object 0", actual.String())
 }
 
+// TestPodPatchUnmarshalling tests the fix for issues #3483, by adding a
+// PodPatch to a pipeline spec and making sure it's applied correctly
+func TestPodPatchUnmarshalling(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	c := getPachClient(t)
+	require.NoError(t, c.DeleteAll())
+
+	// Create input data
+	dataRepo := tu.UniqueString(t.Name() + "-data-")
+	require.NoError(t, c.CreateRepo(dataRepo))
+	_, err := c.PutFile(dataRepo, "master", "file", strings.NewReader("foo"))
+	require.NoError(t, err)
+
+	// create pipeline
+	pipeline := tu.UniqueString("pod-patch-")
+	_, err = c.PpsAPIClient.CreatePipeline(
+		context.Background(),
+		&pps.CreatePipelineRequest{
+			Pipeline: client.NewPipeline(pipeline),
+			Transform: &pps.Transform{
+				Cmd:   []string{"bash"},
+				Stdin: []string{"cp /pfs/in/* /pfs/out"},
+			},
+			Input: &pps.Input{Pfs: &pps.PFSInput{
+				Name: "in", Repo: dataRepo, Glob: "/*",
+			}},
+			PodPatch: `[
+				{
+				  "op": "add",
+				  "path": "/volumes/0",
+				  "value": {
+				    "name": "vol0",
+				    "hostPath": {
+				      "path": "/volumePath"
+				}}}]`,
+		})
+	require.NoError(t, err)
+
+	commitIter, err := c.FlushCommit([]*pfs.Commit{client.NewCommit(dataRepo, "master")}, nil)
+	require.NoError(t, err)
+	commitInfos := collectCommitInfos(t, commitIter)
+	require.Equal(t, 1, len(commitInfos))
+
+	var buf bytes.Buffer
+	require.NoError(t, c.GetFile(commitInfos[0].Commit.Repo.Name, commitInfos[0].Commit.ID, "file", 0, 0, &buf))
+	require.Equal(t, "foo", buf.String())
+
+	pipelineInfo, err := c.InspectPipeline(pipeline)
+	require.NoError(t, err)
+
+	// make sure 'vol0' is correct in the pod spec
+	var volumes []v1.Volume
+	rcName := ppsutil.PipelineRcName(pipelineInfo.Pipeline.Name, pipelineInfo.Version)
+	kubeClient := tu.GetKubeClient(t)
+	require.NoError(t, backoff.Retry(func() error {
+		podList, err := kubeClient.CoreV1().Pods(v1.NamespaceDefault).List(
+			metav1.ListOptions{
+				LabelSelector: metav1.FormatLabelSelector(metav1.SetAsLabelSelector(
+					map[string]string{"app": rcName},
+				)),
+			})
+		if err != nil {
+			return err // retry
+		}
+		if len(podList.Items) != 1 || len(podList.Items[0].Spec.Volumes) == 0 {
+			return fmt.Errorf("could not find volumes for pipeline %s", pipelineInfo.Pipeline.Name)
+		}
+		volumes = podList.Items[0].Spec.Volumes
+		return nil // no more retries
+	}, backoff.NewTestingBackOff()))
+	// Make sure a CPU and Memory request are both set
+	for _, vol := range volumes {
+		require.True(t,
+			vol.VolumeSource.HostPath == nil || vol.VolumeSource.EmptyDir == nil)
+		if vol.Name == "vol0" {
+			require.True(t, vol.VolumeSource.HostPath.Path == "/volumePath")
+		}
+	}
+}
```

Comment:
I'm always a fan of more tests, is the coverage here distinctly different from the `TestPodOpts/Patch` test? And would it maybe make sense to have this test as a subtest of `TestPodOpts`?

Reply:
It might not be that different. The reason this even in this PR is that we had a test, `src/server/pps/server/worker_rc_test.go `, that didn't build. This test is basically that test, moved to `pachyderm_test.go`, and fixed.

When I added the TFJob test to `src/server/pps/server`, I couldn't run it initially because of the broken test that was already in that package. Moving this test here allowed me to run the new TFJob test. I could also just delete this, or leave the old test in `s/s/pps/server` and put the TFJob test in `pachyderm_test.go` (TBH, I didn't even have a great reason for putting the TFJob test under `s/s/pps/server` except that `pachyderm_test.go` is huge and the `TFJob` test seemed new and different).